go/capture/driver/airbyte/driver.go

```diff
@@ -443,49 +297,31 @@ func (d driver) Pull(stream pc.Driver_PullServer) error {
 				}
 			}
 		},
-		// Expect to decode Airbyte messages.
-		connector.NewJSONOutput(
-			func() interface{} { return new(airbyte.Message) },
-			func(i interface{}) error {
-				if rec := i.(*airbyte.Message); rec.Log != nil {
-					logger.Log(airbyteToLogrusLevel(rec.Log.Level), nil, rec.Log.Message)
-				} else if rec.State != nil {
-					return pc.WritePullCheckpoint(stream, &resp,
-						&pf.DriverCheckpoint{
-							DriverCheckpointJson: rec.State.Data,
-							Rfc7396MergePatch:    rec.State.Merge,
-						})
-				} else if rec.Record != nil {
-					if b, ok := streamToBinding[rec.Record.Stream]; ok {
-						return pc.StagePullDocuments(stream, &resp, b, rec.Record.Data)
-					}
-					return fmt.Errorf("connector record with unknown stream %q", rec.Record.Stream)
-				} else {
-					return fmt.Errorf("unexpected connector message: %v", rec)
-				}
-				return nil
+		connector.NewProtoOutput(
+			func() proto.Message { return new(pc.PullResponse) },
+			func(m proto.Message) error {
+				return stream.Send(m.(*pc.PullResponse))
 			},
-			onStdoutDecodeError(logger),
 		),
 		logger,
-	); err != nil {
-		return err
-	}
+	)
 
-	if resp == nil {
-		return nil // Connector flushed prior to exiting. All done.
-	}
+	// TODO: should the new logic send out a final commit if there are pending documents without a checkpoint?
+
+	//if resp == nil {
+	//	return nil // Connector flushed prior to exiting. All done.
+	//}
 
 	// Write a final commit, followed by EOF.
 	// This happens only when a connector writes output and exits _without_
```

Comment:
This is something that I am not very sure about.
It looks to me the go-logic will append an `EOF` to the stream if there are pending messages that are not checked in. This logic was missing in Rust, should this be back?